### DISCUSSION TITLE
Gradle build make error

### DISCUSSION THREAD

**Discussion by mmaryo** [ExpediaGroup/graphql-kotlin#892] [Q&A] [answered]
Hello 

My project 
Kotlin 1.4.0
Quarkus 1.8.1.Final
graphql-kotlin 3.6.4 and 4.0.0-alpha.4

I follow tutorial for create a client in my project 

https://expediagroup.github.io/graphql-kotlin/docs/client/client-overview#build-configuration

I simply add in build.gradle

    implementation("com.expediagroup:graphql-kotlin-ktor-client:$graphqlKotlin")

and 

```
graphql {
    client {
        endpoint = "https://api.thegraph.com/subgraphs/name/uniswap/uniswap-v2"
        packageName = "com.app.uniswap"
        clientType = GraphQLClientType.KTOR
    }
}

```

And I have a file .graphql inside resources folder 

gradle build throw this error 

> Could not set unknown property 'endpoint' for extension 'graphql' of type com.expediagroup.graphql.plugin.gradle.GraphQLPluginExtension.

What is wrong ?

**Comment by dariuszkuc** [comment]
Hello 👋 , can you provide a link to a sample github project that reproduces this issue? Error somewhat implies that you are trying to set `endpoint` property on the `graphql` extension, i.e.

```
// incorrect
graphql {
  endpoint = "whatever"
}

// valid
graphql {
  client {
    endpoint = "whatever"
  }
}
```

**Comment by mmaryo** [comment]
Hello @dariuszkuc 

This is the repo https://github.com/mmaryo/test-graphql-kotlin

and do ./gradlew build

**Comment by dariuszkuc** [comment, marked as answer]
Thanks for raising this, it is indeed a bug on our end (https://github.com/ExpediaGroup/graphql-kotlin/issues/893) - only tested the plugin from Kotlin DSL. TLDR I've used Kotlin lambda which cannot be used from Groovy script so need update the plugin code to make extension useable from both Kotlin and Groovy.

In the meantime you will have to explicitly configure the tasks (or migrate to use KTS)
```
tasks.getByName("graphqlIntrospectSchema") {
    endpoint.set("https://api.thegraph.com/subgraphs/name/uniswap/uniswap-v2")
}
tasks.getByName("graphqlGenerateClient") {
    packageName.set("com.mario.uniswap")
    clientType.set(com.expediagroup.graphql.plugin.generator.GraphQLClientType.KTOR)
    schemaFile.set(graphqlIntrospectSchema.outputFile)
    dependsOn("graphqlIntrospectSchema")
}
```

**Comment by dariuszkuc** [comment]
Extension is fixed in https://github.com/ExpediaGroup/graphql-kotlin/pull/901. PR also updates Gradle plugin docs to reflect both `build.gradle.kts` and `build.gradle` usage.
